### PR TITLE
Added logging to deprecated functions

### PR DESCRIPTION
**Pull request by JP-Lisn**
As discussed in https://github.com/google/oboe/issues/1925, logging when using deprecated functions has been added. I did waffle on if these should be warnings or debug messages... I landed on warnings. I Will change if a different log level is preferred. 

FYI - Just signed the CLA.

### CODE REVIEW

**Review by philburk — APPROVED** [review]
Thanks for doing this!

**Comment by philburk** [comment]
Warnings are good because use of the functions can lead to crashes.

**Comment by JP-Lisn** [comment]
Whoops, let me double check this.

File: include/oboe/AudioStreamBuilder.h [hunk 1]
```diff
@@ -515,6 +518,8 @@ class AudioStreamBuilder : public AudioStreamBase {
      * @return pointer to the builder so calls can be chained
      */
     AudioStreamBuilder *setCallback(AudioStreamCallback *streamCallback) {
+        LOGW("`setCallback` is deprecated, Call setDataCallback(std::shared_ptr<AudioStreamDataCallback>) and
```
Comment (philburk) [inline include/oboe/AudioStreamBuilder.h:521]: Needs a " at the end of 521 and a " and space at the beginning of line 521

**Review by philburk — CHANGES_REQUESTED** [review]
CI caught build errors. Did you get build errors when you tested this?

You need to:

#include "common/OboeDebug.h"

But that might interfere with making the pre-built.
We normally only do logging in .cpp files.

**Comment by JP-Lisn** [comment]
Sorry about that, a little embarrassing 😞 I did run the build script, but I must have accidentally done so from the wrong source directory.

**Comment by JP-Lisn** [comment]
Interesting, 

Both the `build_all_android.sh` and `prefab_build.sh` complete successfully when run locally.

**Review by philburk — CHANGES_REQUESTED** [review]
Moving the OboeDebug.h file is an API change that could break existing application builds.
We will need to discuss this internally and decide how to move forward.
Please hold off on any more changes. Thanks.

**Comment by JP-Lisn** [comment]
Totally understand. Initially, I thought this change was all good until I built the sample applications. Tests and build scripts all worked. The only way I could solve this without code repetition was to expose that header publicly so the linker could find it. I have run all the samples and tests and found nothing that broke thus far with this change.  

I didn't think this would cause issues with existing builds, as the header was not exposed before. I assumed only internal library references would be at issue, although I realize I might be missing something here. 

If Internally you decide on a different path, I would be happy to assist in making that happen :) 
Thanks again for all your help.

**Comment by JP-Lisn** [comment]
Closing, for now replaced with: https://github.com/google/oboe/pull/1930

**Pull request closed without merging**